functorch/csrc/BatchedTensorImpl.cpp

```diff
@@ -71,6 +72,7 @@ void BatchedTensorImpl::refreshSizesAndStrides() {
     sizes_and_strides_.size_at_unchecked(dim) = value_sizes.at(actual_dim);
     sizes_and_strides_.stride_at_unchecked(dim) = value_strides.at(actual_dim);
   }
+  storage_offset_= value_.storage_offset();
```

Comment:
We should rename refreshSizesAndStrides() to refreshTensorMetadata() or something